include/ListView/ListView.php

```diff
@@ -1884,45 +1884,34 @@ function processListViewTwo($seed, $xTemplateSection, $html_varName) {
         }
     }
 
+    /**
+     * @deprecated
+     * @return string
+     */
     function getArrowStart() {
-        $imgFileParts = pathinfo(SugarThemeRegistry::current()->getImageURL("arrow.gif"));
-
-        return "&nbsp;<!--not_in_theme!--><img border='0' src='".$imgFileParts['dirname']."/".$imgFileParts['filename']."";
+        return '';
     }
 
     function getArrowUpDownStart($upDown) {
         $ext = ( SugarThemeRegistry::current()->pngSupport ? "png" : "gif" );
 
-        if (!isset($upDown) || empty($upDown)) {
-            $upDown = "";
+        if ($upDown === '_down') {
+            return '<span class="suitepicon suitepicon-action-sorting-ascending"></span>';
+        } elseif (($upDown === '_up')) {
+            return '<span class="suitepicon suitepicon-action-sorting-descending"></span>';
+        } else {
+            return '<span class="suitepicon suitepicon-action-sorting-none"></span>';
         }
-        return "&nbsp;<img border='0' src='".SugarThemeRegistry::current()->getImageURL("arrow{$upDown}.{$ext}")."' ";
     }
 
 	function getArrowEnd() {
-		$imgFileParts = pathinfo(SugarThemeRegistry::current()->getImageURL("arrow.gif"));
-
-        list($width,$height) = ListView::getArrowImageSize();
-
-		return '.'.$imgFileParts['extension']."' width='$width' height='$height' align='absmiddle' alt=".translate('LBL_SORT').">";
+	    //
+        return '';
```

Comment:
same like in include/ListView/ListView.php:1891

Reply:
btw, why is the function should return an empty string only? can we just remove these function?

Reply:
I don't think we should remove them right away. I think we should mark this code as deprecated first and then remove it at a later release. This code is using the x-template library. This whole class really needs to be removed and/or replaced with the smarty code version. Refactoring the listview is out of scope for this task. We just need to change the images to icons for now.

These methods are public, so i feel we should be hesitant to change the public interfaces. I don't want change the structure of the codebase to much.